src/Infrastructure/Elastic/ElasticIndexAdapter.php

```diff
@@ -22,9 +24,43 @@ public function create(IndexConfigurationInterface $indexConfiguration): void
         $this->client->indices()->create($indexConfiguration->toArray());
     }
 
+    public function pointToAlias(IndexConfigurationInterface $indexConfiguration): void
+    {
+        if (!$indexConfiguration->isAliased()) {
+            return;
+        }
+
+        $this->upsertAlias($indexConfiguration->getAliasConfiguration());
+
+        if($indexConfiguration->getAliasConfiguration()->shouldOldAliasesBePruned()) {
+            $this->pruneAliases($indexConfiguration->getAliasConfiguration());
+        }
```

Comment:
Should this be part of this function? It seems like a weird side effect to have  in a function called "pointToAlias" . I think it is nicer to put this logic of "pruning the alias after updating the alias" in the ElasticEngine.

Reply:
There are multiple ways/reasons to use aliases. With `pointToAlias` I only intend to use/support the "zero downtime swapping" way of using them. This means that an alias should always end up with only one index in it. Another use would mean adding a function `addIndexToAlias`.